sys/net/gnrc/pkt/gnrc_pkt.c

```diff
@@ -11,22 +12,18 @@
  *
  * @file
  * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author  Sebastian Meiling <s@mlng.net>
  */
 
-#include <assert.h>
-
 #include "net/gnrc/pkt.h"
 
-gnrc_pktsnip_t *gnrc_pktsnip_search_type(gnrc_pktsnip_t *ptr,
+gnrc_pktsnip_t *gnrc_pktsnip_search_type(gnrc_pktsnip_t *pkt,
                                          gnrc_nettype_t type)
 {
-    while (ptr != NULL) {
-        if (ptr->type == type) {
-            return ptr;
-        }
-        ptr = ptr->next;
+    while (pkt && (pkt->type != type)) {
```

Comment:
I would still prefer `(pkt != NULL)` instead of just `pkt`, to prevent type-mixing.